overlord/devicestate/handlers_install.go

```diff
@@ -92,3 +114,31 @@ func (m *DeviceManager) doSetupRunSystem(t *state.Task, _ *tomb.Tomb) error {
 
 	return nil
 }
+
+// TODO:UC20: set to real TPM availability check function
+var checkTPMAvailability = func() error {
+	return nil
+}
+
+func checkEncryption(model *asserts.Model) (res bool, err error) {
+	secured := model.Grade() == asserts.ModelSecured
+	signed := model.Grade() == asserts.ModelSigned
+
+	// check if we should disable encryption non-secured devices
+	if osutil.FileExists(filepath.Join(dirs.RunMnt, "ubuntu-seed", ".force-unencrypted")) {
+		if secured || signed {
+			err = fmt.Errorf("cannot bypass encryption in a secured or signed device")
```

Comment:
I don't think we should error if the file is there in these cases, we should continue and ignore it

Reply:
to be more concrete, I mean:

```
if dangerous && osutil.FileExists(filepath.Join(dirs.RunMnt, "ubuntu-seed", ".force-unencrypted")) {
    return false, nil
}
```

Reply:
Logic changed as suggested.